tools/check_dist.py

```diff
@@ -28,13 +28,10 @@ def ensure_all_artifacts_included(pyproject, wheel_paths):
 
 
 def main(argv):
-    parser = argparse.ArgumentParser("check_dist", exit_on_error=False)
+    parser = argparse.ArgumentParser("check_dist")
     parser.add_argument("pyproject", type=Path)
     parser.add_argument("wheels", type=Path, nargs="*")
-    try:
-        args = parser.parse_args(argv)
-    except SystemExit:
```

Comment:
I wasn't exactly sure about this. `exit_on_error=False` seemed to suggest that a system exit was unwanted, but the test `test_usage` suggested the exit behavior was wanted (only with exit_on_error the usage message is printed). Python 3.13 fixed a bug where exit_on_error=False wasn't respected, so now the test failed. I opted for using the test as expected behavior. @niklasmohrin?